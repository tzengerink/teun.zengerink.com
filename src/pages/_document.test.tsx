import { JSDOM } from 'jsdom'
import { render } from '@testing-library/react'
import { DocumentProps } from 'next/document'
import React from 'react'
import Document from './_document.page'

jest.mock('next/document', () => ({
  __esModule: true,
  default: jest.requireActual('next/document').default,
  Html: ({ children }) => <html>{children}</html>,
  Head: ({ children }) => <head>{children}</head>,
  Main: () => <>[MAIN]</>,
  NextScript: () => <>[NEXTSCRIPT]</>,
}))

describe('Document', () => {
  const ENV = process.env

  beforeEach(() => {
    jest.resetModules()
    process.env = { ...ENV }
  })

  afterAll(() => {
    process.env = ENV
  })

  it('renders correctly', () => {
    process.env.NEXT_PUBLIC_TRACKING_ID = ''
    const dom = new JSDOM()
    const props = {} as DocumentProps
    const { container } = render(<Document {...props} />, { container: dom.window.document })
    expect(container.documentElement).toMatchSnapshot()
  })

  it('renders script tags when tracking id is set', () => {
    const dom = new JSDOM()
    const props = {} as DocumentProps
    const { container } = render(<Document {...props} />, { container: dom.window.document })
    expect(container.documentElement).toMatchSnapshot()
  })
})
