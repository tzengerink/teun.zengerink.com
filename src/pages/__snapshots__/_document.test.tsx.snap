// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document renders empty head when tracking id not set 1`] = `
<html>
  <head />
  <body
    class="bg-white m-0"
  >
    [MAIN]
    [NEXTSCRIPT]
  </body>
</html>
`;

exports[`Document renders script tags when tracking id is set 1`] = `
<html>
  <head>
    <script
      async=""
      src="https://www.googletagmanager.com/gtag/js?id=ga-tracking-id"
    />
    <script>
      
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
      gtag('config', 'ga-tracking-id', {
        page_path: window.location.pathname,
      });
    </script>
  </head>
  <body
    class="bg-white m-0"
  >
    [MAIN]
    [NEXTSCRIPT]
  </body>
</html>
`;
